### ISSUE TITLE
Share Button Not Working

### ISSUE THREAD

**Issue by TedEngel** [ZachSaucier/Just-Read#218]
Hi -- I'm a long-time JustRead user who has finally taken the step of purchasing the premium version. Everything works as I expect other than the Share button, which doesn't do anything when I click it.

I'm on the latest version of JustRead Premium (downloaded and installed a few minutes ago). I'm also using  Windows 7 Enterprise, Service Pack 1.

**Comment by ZachSaucier** [comment]
So sorry to hear that! Thanks for the report and for using Just Read for a while! I'm glad you like it.

Can you open up the developer console (pressing F12 is probably the easiest way) and then try sharing the page and take a screenshot of any errors that appear?

**Comment by TedEngel** [comment]
Hi Zach -- I'm not a developer, but everything else associated with the Premium version other than the Share button seems to be working well.  Since my original message, I've downloaded and installed the latest version of Chrome and I also visited a variety of sites to test the extension.

I selected an article from the Longreads website. Like before, the button is unresponsive other than the rollover effect. I captured the following screenshot for your review.

![screenshot-1](https://user-images.githubusercontent.com/45859931/50021435-b9af7700-ff9e-11e8-9b34-ffa15f1160da.jpg)

**Comment by ZachSaucier** [comment]
Ya did great! The "Empty UID. Failing to share page." error message in the bottom right is the problematic one. Are you signed into multiple Google accounts by chance?

**Comment by dkthrive** [comment]
New user (1 day) but appear to have same problem. Just so I understand - I am trying to send full text Just-Read article and nothing happens when I hit share button.
![just-read-1](https://user-images.githubusercontent.com/45965107/50146264-5c0e7980-0281-11e9-9345-17facef35e02.PNG)

**Comment by dkthrive** [comment]
> New user (1 day) but appear to have same problem. Just so I understand - I am trying to send full text Just-Read article and nothing happens when I hit share button.
> ![just-read-1](https://user-images.githubusercontent.com/45965107/50146264-5c0e7980-0281-11e9-9345-17facef35e02.PNG)

and should "Share" function send full text as PDF or in-line HTML or text within email?

**Comment by ZachSaucier** [comment]
The share functionality creates a new HTML webpage hosted on justread.link

@dkthrive can you please click the "console" tab of the developer tools to see if it's the same error causing the issue for your? And are you signed into multiple Google accounts by chance?

**Comment by dkthrive** [comment]
Hi Zach. Thanks for responding.  Console tab on new Justread share is blank.  See attached:
![just-read-2](https://user-images.githubusercontent.com/45965107/50164499-e4f0d980-02b0-11e9-9d8d-2a69bcfc3740.PNG)

And no, I do not believe that I am signed into multiple Google accounts.

**Comment by dkthrive** [comment]
Zach, if I were to share justread via gmail, would I be able to share all of the text in the article or just your link such as ​https://justread.link/PSval5DC-

**Comment by dkthrive** [comment]
> Zach, if I were to share justread via gmail, would I be able to share all of the text in the article or just your link such as ​https://justread.link/PSval5DC-

Because I was looking for a full text share similar to what is available on IOS/Mac. If this is not available, this app won't work for me. Let me know. Thanks.

**Comment by ZachSaucier** [comment]
@dkthrive That is functioning as it is meant to function. But the functionality that you're suggesting - emailing the text content to another person - is an interesting idea I hadn't thought of and I'll look into implementing it now! I created [a new issue](https://github.com/ZachSaucier/Just-Read/issues/220) for that feature request, so please post any further comments there.

**Comment by dkthrive** [comment]
Thanks Zach. I was looking for full share.  Can you please refund my app fee as your app isn't going to meet my needs.

**Comment by TedEngel** [comment]
Hi Zach -- Sorry for the delay in responding. I was signed into two Google accounts at one time (one at work, one at home), but I have returned to my main Google account and I still have the same issue.  I appreciate the help and love the extension, and I'm happy to wait to see how we can get this right.

**Comment by TedEngel** [comment]
![tedengel--2](https://user-images.githubusercontent.com/45859931/50182754-c94bfa00-02d5-11e9-9d6c-b32752ae1060.png)

**Comment by ZachSaucier** [comment]
@dkthrive As the [end user license agreement (EULA)](https://github.com/ZachSaucier/Just-Read/blob/master/docs/EULA.md#no-warranty) for JR says, refunds are not given for Just Read  Premium. 

I'll continue investigating the feasibility of sending articles over email using JRP. Until then, you can share the article text by sharing the generated justread.link URL of the article (so send https://justread.link/PSval5DC- to people over email for example).

**Comment by ZachSaucier** [comment]
@TedEngel Thanks for the information.

Can you try running the following in your console and emailing me the response (**Note** that the response has sensitive information, so please don't post it on this issue)? It should be a number. My email is support@justread.link

`chrome.runtime.sendMessage('dpamdgmnffodphoamchpmbclkpmccjga', { getUID: "true" }, function(response) { console.log(response.uid); });`

**Comment by dkthrive** [comment]
> @dkthrive As the [end user license agreement (EULA)](https://github.com/ZachSaucier/Just-Read/blob/master/docs/EULA.md#no-warranty) for JR says, refunds are not given for Just Read Premium.
> 
> I'll continue investigating the feasibility of sending articles over email using JRP. Until then, you can share the article text by sharing the generated justread.link URL of the article (so send https://justread.link/PSval5DC- to people over email for example).

Really?  I will just dispute the charge with my card company

**Comment by TedEngel** [comment]
> @TedEngel Thanks for the information.
> 
> Can you try running the following in your console and emailing me the response (**Note** that the response has sensitive information, so please don't post it on this issue)? It should be a number. My email is [support@justread.link](mailto:support@justread.link)
> 
> `chrome.runtime.sendMessage('dpamdgmnffodphoamchpmbclkpmccjga', { getUID: "true" }, function(response) { console.log(response.uid); });`

Could you provide some more information? I'm happy to help, but I don't fully understand your request.

**Comment by ZachSaucier** [comment]
@TedEngel Copy the gray lines in my post above, then paste them in the "Console" section of the developer tools (where you saw the errors before). It should return a number (which is a private number - please don't share it in this email/GitHub issue!). Send that number via a separate email to support@justread.link and I'll investigate the issue using that information. 

It's possible that it will return a blank, empty string. If so, please just tell me that that happened.

**Comment by TedEngel** [comment]
Just sent you a screen capture.

**Comment by ZachSaucier** [comment]
Thanks for that information. 

Can you try navigating to chrome://extensions/?id=dpamdgmnffodphoamchpmbclkpmccjga and see if Just Read shows? If so, what's the version number?

**Comment by TedEngel** [comment]
I see Just Read Premium, Version 3.2.3.

**Comment by ZachSaucier** [comment]
Interesting. I'm going to contact you over email to see if we can investigate your situation more.

**Comment by knightwolfjk** [comment]
I'm here by way of the Support link on the JustRead page.  Long time fan, first time caller.  Anything anyone else can do to assist?

**Comment by ZachSaucier** [comment]
@knightwolfjk Can you see what the console says, if anything, about the issue? Does it happen on every page you try? Has the share functionality worked in the past?

**Comment by ZachSaucier** [comment]
I'm closing this issue as I don't think there's anything in particular that I can test right now. Happy to reopen it if someone provides steps to reproduce it.

**Comment by Gnarly12** [comment]
I seem to be having the same problem @ZachSaucier 

Been using justread for a while and just bought premium but share button is performing as for TedEngel.

Did you guys find a resolution? Thanks.

**Comment by ZachSaucier** [comment]
@Gnarly12 You're receiving the "Empty UID. Failing to share page." error?

**Comment by Gnarly12** [comment]
@ZachSaucier I'm not very code savvy but I tried the F12 thing you mentioned earlier and did a screen record: https://cl.ly/bdcb731ab0d4

Let me know if I need to adjust something so you can see what you need.

**Comment by ZachSaucier** [comment]
@Gnarly12 See the "Console" tab next to "What's New"? Click on that and let me know what is in the screen that comes up when you click on it.

**Comment by Gnarly12** [comment]
https://cl.ly/440c3e4ae41e

**Comment by ZachSaucier** [comment]
@Gnarly12 Would you mind please sending me an email at support@justread.link? I'd like to ask some additional questions that may require a little bit of sensitive information.

**Comment by Gnarly12** [comment]
Done.

**Comment by ZachSaucier** [comment]
Most likely this is error caused by not enabling user account syncing on your Google account. To enable syncing, go to Chrome's settings page (chrome://settings/) and under "People" click "Turn on syncing". 

___

Thanks a lot to @Gnarly12 for helping me figure out the cause of this issue! I've updated [the FAQ](https://github.com/ZachSaucier/Just-Read/#q-how-can-i-share-my-just-read-version-of-a-page-with-someone) with this necessary setting.

**Comment by photonutt** [comment]
Same issue here.  Non-working Share button.  This is a very large part of the reason I purchased the Pro version.  Hoping there is a solution for this?

Here is what I am seeing:

cdn.onesignal.com/sdks/OneSignalSDK.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
jquery-migrate.min.js?ver=1.4.1:2 JQMIGRATE: Migrate is installed, version 1.4.1
www.google-analytics.com/analytics.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
stats.wp.com/e-201932.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
secure.quantserve.com/quant.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
pixel.wp.com/g.gif?v=wpcom-no-pv&x_sharing-count-request=pinterest&r=0.28115916850025946:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
pixel.wp.com/g.gif?v=wpcom-no-pv&x_sharing-count-request=facebook&r=0.3895186395562851:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
the-muslims-inner-struggles:1 Unchecked runtime.lastError: The message port closed before a response was received.
s2.wp.com/:8 JQMIGRATE: Migrate is installed, version 1.4.1
??-eJyFztEKwjAMBdAfsisyOnwRv6XWOFKXtDbphn69HeiDMBQC9+EeuLFLNshhqhcQG9vdK5THO7ooO/sLGMKxeIWOkD84JFZgXW1OogQifoSNltIZJzBVoDTA2mauacN9v4Q8Iyx/WQTNPtxMAcHnun6i4965Q++GwfXxBTkPW9k=:8 JQMIGRATE: Migrate is installed, version 1.4.1
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
chrome-extension://invalid/:1 Failed to load resource: net::ERR_FAILED
7content_script.js:2031 Empty UID. Failing to share page. This may be caused by not enabling user account syncing under Chrome's settings page. For more information, please see https://github.com/ZachSaucier/Just-Read/#q-how-can-i-share-my-just-read-version-of-a-page-with-someone
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
4content_script.js:2031 Empty UID. Failing to share page. This may be caused by not enabling user account syncing under Chrome's settings page. For more information, please see https://github.com/ZachSaucier/Just-Read/#q-how-can-i-share-my-just-read-version-of-a-page-with-someone
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.
H6kWrI0V0LQ?version=3&rel=1&fs=1&autohide=2&showsearch=0&showinfo=1&iv_load_policy=1&wmode=transparent:1 Access to XMLHttpRequest at 'https://googleads.g.doubleclick.net/pagead/id' from origin 'https://www.youtube.com' has been blocked by CORS policy: The value of the 'Access-Control-Allow-Origin' header in the response must not be the wildcard '*' when the request's credentials mode is 'include'. The credentials mode of requests initiated by the XMLHttpRequest is controlled by the withCredentials attribute.

**Comment by ZachSaucier** [comment]
Hey @photonutt sorry to hear about your issue.

The only error related to Just Read in that list that you provided is 

```
Empty UID. Failing to share page. This may be caused by not enabling user account syncing under Chrome's settings page. For more information, please see https://github.com/ZachSaucier/Just-Read/#q-how-can-i-share-my-just-read-version-of-a-page-with-someone
```

Have you read [the linked FAQ question](https://github.com/ZachSaucier/Just-Read/#q-how-can-i-share-my-just-read-version-of-a-page-with-someone) and tried what it said? 

If so and it's still occurring, please let me know if this happens on all sites or specific ones and any other information about your environment that may be relevant.

**Comment by photonutt** [comment]
That is probably the problem then.  I use Brave browser https://brave.com/

It uses all of Chrome's extensions and up until now I have not had any
trouble with them, but perhaps the share function simply doesn't work with
it?

**Comment by ZachSaucier** [comment]
@photonutt I'd guess that that is indeed the issue. Brave does about all that it can to prevent syncing.

How about you send an email to support@justread.link and I can see if I can hook you up with a workaround?
